homeassistant/auth/session.py

```diff
@@ -0,0 +1,197 @@
+"""Session auth module."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import secrets
+from typing import TYPE_CHECKING, TypedDict
+
+from aiohttp.web import Request
+from aiohttp_session import Session, get_session, new_session
+from cryptography.fernet import Fernet
+
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.storage import Store
+from homeassistant.util import dt as dt_util
+
+from .models import RefreshToken
+
+if TYPE_CHECKING:
+    from . import AuthManager
+
+
+TEMP_TIMEOUT = timedelta(minutes=5)
+TEMP_TIMEOUT_SECONDS = timedelta(minutes=5).total_seconds()
+
+SESSION_ID = "id"
+STORAGE_VERSION = 1
+STORAGE_KEY = "auth.session"
+
+
+@dataclass
+class StrictConnectionTempSessionData:
+    """Session data for accessing unauthorized resources for a short period of time, when strict connection is enabled."""
+
+    cancel_remove: CALLBACK_TYPE
+    absolute_expiry: datetime = dt_util.utcnow() + TEMP_TIMEOUT
```

Comment:
Why do we set the default to import time + offset? Why do we never set this attribute during run?

Reply:
Thanks for the catch. I use now the init function